Add 1040 form to datasets page in docs (#214)
ihmeuw/pseudopeople: docs/source/datasets/index.rst

     - Rounded down to an integer.
   * - Date of birth
     - :code:`date_of_birth`

>      - Formatted as MMDDYYYY.

Reviewer: These date formatting changes shouldn't be in this PR, as they are covered in #216.

Author: Ooh okay idk why I got nervous there would be a merge conflict so I thought this would be proactive but I can undo them..

.. list-table:: **Dataset columns**
   :header-rows: 1

   * - Attribute Name
     - Column Name
     - Notes
   * - Unique simulant ID
     - :code:`simulant_id`
     - Not affected by noise functions; intended use is "ground truth" for testing and validation.
   * - Unique household ID
     - :code:`household_id`
     - Not affected by noise functions; intended use is "ground truth" for testing and validation; consistent across all
       datasets. 
   * - First name
     - :code:`first_name`
     - 
   * - Middle initial
     - :code:`middle_initial`
     - 
   * - Last name
     - :code:`last_name`
     - 
   * - Mailing address street number
     - :code:`mailing_address_street_number`
     - 
   * - Mailing address street name
     - :code:`mailing_address_street_name`
     - 
   * - Mailing address unit number
     - :code:`mailing_address_unit_number`
     - 
   * - Mailing address PO box
     - :code:`mailing_address_po_box`
     - 
   * - Mailing address city
     - :code:`mailing_address_city`    
     - 
   * - Mailing address state
     - :code:`mailing_address_state`  
     - 
   * - Mailing address ZIP code
     - :code:`mailing_address_zipcode`
     - 

>    * - Social security number 

Reviewer: ```suggestion
   * - Social Security Number (SSN)
```

Since this acronym is used both in the notes column, and in the column name itself

     - :code:`ssn`
     - Individual Taxpayer Identification Number (ITIN) if no SSN
   * - Income 
     - :code:`income`
     - 
   * - Employer ID 
     - :code:`employer_id`
     -  

>    * - Employer Name 

Reviewer: ```suggestion
   * - Employer name
```

     - :code:`employer_name`
     - 
   * - Employer street number
     - :code:`employer_street_number`
     - 
   * - Employer street name
     - :code:`employer_street_name`
     - 
   * - Employer unit number
     - :code:`employer_unit_number`
     - 
   * - Employer city
     - :code:`employer_city`    
     - 
   * - Employer state
     - :code:`employer_state`  
     - 
   * - Employer ZIP code
     - :code:`employer_zipcode`
     - 

>    * - Joint filer first name

Reviewer: Is there a reason why we say "joint filer" in this table but "spouse" in the column name itself? I guess "spouse" is a little more confusing (you could have a spouse that doesn't appear in your 1040, if you file separately) so we opted for a more precise term but didn't want to bug the engineers?

Author: I guess so - I think that's what was decided in this [Slack thread ](https://ihme.slack.com/archives/C02KUQ9LX32/p1688405675889299)

     - :code:`spouse_first_name`
     - 
   * - Joint filer middle initial
     - :code:`spouse_middle_initial`
     - 
   * - Joint filer last name
     - :code:`spouse_last_name`
     - 
   * - Joint filer social security number
     - :code:`spouse_ssn`
     - Individual Taxpayer Identification Number (ITIN) if no SSN
   * - Joint filer income
     - :code:`spouse_income`
     - 
   * - Joint filer employer ID
     - :code:`spouse_employer_id`
     - 
   * - Joint filer employer name
     - :code:`spouse_employer_name`
     - 
   * - Joint filer employer address
     - :code:`spouse_employer_address`
     - 
   * - Joint filer employer ZIP code
     - :code:`spouse_employer_zipcode`
     - 
   * - Dependent 1 first name
     - :code:`dependent_1_first_name`
     - 
   * - Dependent 1 last name
     - :code:`dependent_1_last_name`
     - 

>    * - Dependent 1 social security number

Reviewer: ```suggestion
   * - Dependent 1 Social Security Number (SSN)
```

You could do this, **or** use just "SSN" since you will have introduced that acronym above. I kind of like spelling it out again, but I don't feel strongly.

Tax form: 1040
--------------
As with data collected from W-2 and 1099 forms, Pseudopeople will also enable the simulation of administrative records from 1040 forms, which are
also reported to the IRS on an annual basis. To find out more about the 1040 tax form, visit the `IRS information page <https://www.irs.gov/instructions/i1040gi>`_.

> Generate 1040 data with pseudopeople.generate_taxes_1040.

Reviewer 2: ```suggestion
Generate 1040 data with  :func:`pseudopeople.generate_taxes_1040`.
```

Author: This is what I had originally, but I think bc the docstring hasn't been created yet, git flagged it and failed the check

Reviewer 3: Maybe we should add a to-do in the docs, or at least a JIRA ticket, to update this once the code is written

The following columns are included in these datasets:

.. list-table:: **Dataset columns**
   :header-rows: 1

   * - Attribute Name
     - Column Name
     - Notes
   * - Unique simulant ID
     - :code:`simulant_id`
     - Not affected by noise functions; intended use is "ground truth" for testing and validation.
   * - Unique household ID
     - :code:`household_id`
     - Not affected by noise functions; intended use is "ground truth" for testing and validation; consistent across all
       datasets. 
   * - First name
     - :code:`first_name`
     - 
   * - Middle initial
     - :code:`middle_initial`
     - 
   * - Last name
     - :code:`last_name`
     - 
   * - Age
     - :code:`age`  
     - Rounded down to an integer.
   * - Date of birth
     - :code:`date_of_birth`

>      - Formatted as YYYY-MM-DD.

Reviewer 2: double check that this is the format we want; i could imagine MM/DD/YYYY being more appropriate, but I don't mind whatever we choose.

Reviewer 3: Age and DOB [should not appear at all in tax forms](https://github.com/ihmeuw/vivarium_research/pull/1235). The "Age" and "Date of birth" rows should be removed.

Reviewer 3: According to the [pseudopeople docs](https://pseudopeople.readthedocs.io/en/latest/datasets/index.html), all of our other datasets have DOB in YYYY-MM-DD format as well (Census, ACS, CPS, WIC, SSA). @zmbc does that seem accurate?

   * - Mailing address street number
     - :code:`mailing_address_street_number`
     - 
   * - Mailing address street name
     - :code:`mailing_address_street_name`
     - 
   * - Mailing address unit number
     - :code:`mailing_address_unit_number`
     - 
   * - Mailing address PO box
     - :code:`mailing_address_po_box`
     - 
   * - Mailing address city
     - :code:`mailing_address_city`    
     - 
   * - Mailing address state
     - :code:`mailing_address_state`  
     - 
   * - Mailing address ZIP code
     - :code:`mailing_address_zipcode`
     - 
   * - Social security number 
     - :code:`ssn`
     - 

>    * - Individual tax identification number

Reviewer 3: According to [Zeb](https://ihme.slack.com/archives/C02KUQ9LX32/p1688762729760379?thread_ts=1688762551.503559&cid=C02KUQ9LX32) and to [the 1040 form itself](https://www.irs.gov/pub/irs-pdf/f1040.pdf), we shouldn't have a separate column for ITIN. Instead ITIN's should go in the SSN column, and we should add a note to that column saying that values in that column will be ITIN's for people who use those instead of SSN's.

     - 
   * - Employer city
     - :code:`employer_city`    
     - 
   * - Employer state
     - :code:`employer_state`  
     - 
   * - Employer ZIP code
     - :code:`employer_zipcode`
     - 
   * - Joint filer first name
     - :code:`spouse_first_name`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer middle initial
     - :code:`spouse_middle_initial`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer last name
     - :code:`spouse_last_name`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer mailing address street number
     - :code:`spouse_mailing_address_street_number`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer mailing address street name
     - :code:`spouse_mailing_address_street_name`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer mailing address unit number
     - :code:`spouse_mailing_address_unit_number`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer mailing address PO box
     - :code:`spouse_mailing_address_po_box`

>      - Only included if the simulant has a joint filer (i.e., spouse).

Reviewer 3: The [actual 1040 form](https://www.irs.gov/pub/irs-pdf/f1040.pdf) only has space for one address, so we should remove the separate address columns for the joint filer [as agreed on Slack](https://ihme.slack.com/archives/C02KUQ9LX32/p1688764040017429?thread_ts=1688762551.503559&cid=C02KUQ9LX32).

   * - Joint filer social security number
     - :code:`spouse_ssn`
     - Only included if the simulant has a joint filer (i.e., spouse).

>    * - Joint filer individual tax identification number

Reviewer 3: Same comment as above -- ITINs should go in the SSN column and this column should be removed.

     - :code:`spouse_itin`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer income
     - :code:`spouse_income`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer employer ID
     - :code:`spouse_employer_id`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer employer name
     - :code:`spouse_employer_name`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer employer address
     - :code:`spouse_employer_address`
     - Only included if the simulant has a joint filer (i.e., spouse).
   * - Joint filer employer ZIP code
     - :code:`spouse_employer_zipcode`

>      - Only included if the simulant has a joint filer (i.e., spouse).

Reviewer 2: You say "spouse" and "joint filer" a lot... do we need to include both of these terms?

Reviewer 3: I think the "only included if" wording is unclear. The DataFrame will still have these columns regardless of whether a simulant has a joint filer or not (same for dependents), but they will just be filled with NaN's in rows where they don't apply. I propose one of two things:

1. Just delete these notes. That would agree with what we've done for other values that may be missing, such as unit number.
2. Change the notes to something like "NaN if simulant does not file jointly" or "NaN if simulant has no dependents" or "NaN if simulant has fewer than 2 dependents," etc.

   * - Dependent 1 first name
     - :code:`dependent_1_first_name`

>      - Only included if the simulant has a tracked dependent; to be included for each dependent on the tax filing, up to 4 dependents.

Reviewer 2: Annoying as it might be, perhaps you should include dependent_2, 3, and 4 in this same format.  If the engineers don't need you to, don't!

Reviewer 3: Well, _somewhere_ we should indicate that the dataframe has columns all the way through `dependent_4`. We should either just list them all in the same format as Abie says, or maybe we can come up with a clear way to indicate this without having to list them all out...

   * - Dependent 1 middle initial
     - :code:`dependent_1_middle_initial`

>      - Only included if the simulant has a tracked dependent.

Reviewer 3: The term "tracked dependent" is specific to our simulation and won't be meaningful to pseudopeople users. As per my comment above, maybe we should just delete these notes, which would make this a moot point.

   * - Dependent 1 last name
     - :code:`dependent_1_last_name`
     - Only included if the simulant has a tracked dependent.
   * - Dependent 1 social security number
     - :code:`dependent_1_ssn`
     - Only included if the simulant has a tracked dependent.

>    * - Dependent 1 individual tax identification number

Reviewer 3: Another ITIN column to remove and merge into SSN column.

Tax form: 1040
--------------

> As with data collected from W-2 and 1099 forms, Pseudopeople will also enable the simulation of administrative records from 1040 forms, which are

Reviewer: ```suggestion
As with data collected from W-2 and 1099 forms, pseudopeople enables the simulation of administrative records from 1040 forms, which are
```

also reported to the IRS on an annual basis. To find out more about the 1040 tax form, visit the `IRS information page <https://www.irs.gov/instructions/i1040gi>`_.

Generate 1040 data with pseudopeople.generate_taxes_1040.

.. todo::
  Add link to docstring once ready.

> The following columns are included in these datasets:

Reviewer: ```suggestion
The following columns are included in this dataset:
```

Or am I missing something here?
